packages/components/src/Disclosure/Disclosure.tsx

```diff
@@ -67,15 +64,12 @@ export function Disclosure({
     </details>
   );
 
-  function onToggle(event: React.MouseEvent<HTMLDetailsElement>) {
-    event.preventDefault();
+  function handleToggle(event: React.MouseEvent<HTMLDetailsElement>) {
+    console.log("handleToggle");
     const { open: currentToggleState } = event.target as HTMLDetailsElement;
-
-    if (!isMounted || currentToggleState === isOpen) {
-      return;
-    }
-
-    setOpen(!currentToggleState);
+    if (currentToggleState === isOpen) return;
+    setInternalIsOpen(() => !isOpen);
+    onToggle?.(!isOpen);
```

Comment:
something odd is happening, it looks like even though it is controlled the disclosure still toggles even if `opened` doesn't change. Going to look into this

Reply:
This looks to be because of [this](https://github.com/facebook/react/issues/15486) issue. I have an idea that I will test out